std-bits/table/src/main/java/org/enso/table/data/table/TableToXml.java

```diff
@@ -0,0 +1,102 @@
+package org.enso.table.data.table;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.apache.xmlbeans.XmlException;
+import org.apache.xmlbeans.XmlObject;
+import org.apache.xmlbeans.XmlOptions;
+import org.graalvm.polyglot.Context;
+import org.w3c.dom.DOMException;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+public class TableToXml {
+
+  public static String to_xml(
+      int rowCount,
+      Column[] element_columns,
+      Column[] attribute_columns,
+      Column value_Column,
+      String root_name,
+      String row_name)
+      throws XmlException, ParserConfigurationException {
+    var docFactory = DocumentBuilderFactory.newInstance();
+    var docBuilder = docFactory.newDocumentBuilder();
+    var doc = docBuilder.newDocument();
+    var rootElement =
+        doc.createElement(root_name.isEmpty() ? "root" : makeXmlTagNameLegal(root_name));
+    doc.appendChild(rootElement);
+
+    var context = Context.getCurrent();
+    for (int row = 0; row < rowCount; row++) {
+      var rowElement =
+          doc.createElement(row_name.isEmpty() ? "row" : makeXmlTagNameLegal(row_name));
```

Comment:
no need to normalise the row_name on every single row. Should do as precursor step.